torch/_inductor/cudagraph_trees.py

```diff
@@ -1196,11 +1235,27 @@ def new_graph_id(self) -> GraphID:
     def new_func_id(self) -> FunctionID:
         return FunctionID(next(self.func_counter))
 
-    def add_function(self, model, inputs, static_input_idxs) -> Callable:
+    def add_function(
+        self,
+        model,
+        inputs,
+        static_input_idxs,
+        is_backward,
+        is_inference,
```

Comment:
Why doesn't this use the enum? As is you have the impossible state is backward AND is inference